### PR TITLE
Deprecate cpp-taskflow

### PR DESCRIPTION
**Pull request by intelligide**
- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Waiting for the #1990

### CODE REVIEW

**Comment by uilianries** [comment]
@danimtb @SSE4 @jgsogo Can we use alias instead of keeping all copies? Any bugfix in future shall be replicated for both recipes ...

**Comment by Croydon** [comment]
@uilianries I'm not sure we should do this. Wouldn't it better to get it uniform by getting people to migrate once to the new name?

**Comment by uilianries** [comment]
@Croydon I agree, but what about the old versions? Duplication also is not good.

File: recipes/cpp-taskflow/all/conanfile.py [hunk 1]
```diff
@@ -15,10 +15,12 @@ class CppTaskflowConan(ConanFile):
     no_copy_source = True
 
     settings = "os", "compiler"
-    
+
     _source_subfolder = "source_subfolder"
-    
+
     def configure(self):
+        self.output.warn("[DEPRECATED] Package cpp-taskflow is being deprecated. Change yours to require taskflow instead")
```
Comment (jgsogo) [inline recipes/cpp-taskflow/all/conanfile.py:22]: I'd add the reason, something like _"Project upstream has been renamed"_

**Comment by jgsogo** [comment]
This deprecation notice should be enough, we no longer need to make changes to this recipe, consumers should use the new name `taskflow`, if the project has been renamed there is no reason to keep using this one.

Nevertheless, I still think we need something else to manage these recipes, no idea what it is, no idea if it is something built-in in Conan. The main problem is that different versions of `cpp-taskflow` and `taskflow` won't conflict in a graph.

**Comment by IceflowRE** [comment]
https://github.com/conan-io/conan-center-index/pull/2079

**Pull request closed without merging**